parsers/NTESMO.py

```diff
@@ -0,0 +1,214 @@
+        data_points.append(data_point)
+    return data_points
+
+
+def parse_production_mix(
+    raw_production_mix: pd.DataFrame, logger: Logger
+) -> List[dict]:
+    production_mix = []
+    generation_units = set(raw_production_mix.columns)
+    generation_units.remove("Period Start")
+    generation_units.remove("Period End")
+    if not generation_units == PLANT_MAPPING.keys():
+        raise Exception(
+            f"New generator {generation_units - PLANT_MAPPING.keys()} detected in AU-NT, please update the mapping of generators."
+        )
+    for _, production in raw_production_mix.iterrows():
+        data_point = {
+            "zoneKey": "AU-NT",
+            "datetime": production["Period Start"],
+            "source": "ntesmo.com.au",
+            "production": dict(),
+            "storage": dict(),
+        }
+        for generator_key, generator in PLANT_MAPPING.items():
+            if generator_key not in production:
+                raise Exception(
+                    f"Missing generator {generator_key} detected in AU-NT, please update the mapping of generators."
+                )
+            # Some decomissioned plants have negative production values.
+            if production[generator_key] >= 0:
+                if generator["fuel_type"] in data_point["production"]:
+                    data_point["production"][generator["fuel_type"]] += production[
+                        generator_key
+                    ]
+                else:
+                    data_point["production"][generator["fuel_type"]] = production[
+                        generator_key
+                    ]
+        production_mix.append(data_point)
+    return production_mix
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_consumption(
+    zone_key: str = "AU-NT",
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+):
+    consumption = get_data(session, target_datetime, extract_demand_price_data, logger)
+    return parse_consumption(consumption, target_datetime, logger)
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_price(
+    zone_key: str = "AU-NT",
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+):
+    consumption = get_data(session, target_datetime, extract_demand_price_data, logger)
+    return parse_consumption(consumption, target_datetime, logger, price=True)
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_production_mix(
+    zone_key: str = "AU-NT",
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+):
+    production_mix = get_data(session, target_datetime, extract_production_data, logger)
+    return parse_production_mix(production_mix, logger)
+
+
+if __name__ == "__main__":
+    target_datetime = datetime.now() - timedelta(days=2)
```

Comment:
Is this just for testing purposes or is there a 48 hour delay on the data?
If there is a delay on the data we should add the following to the `AU-NT.yaml` file:
```yaml
delays:
  production: 48
  consumption: 48
  # etc
```

Reply:
Good catch, there's indeed a 24 hours delay for the data to get published.